gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilPlugin.kt

```diff
@@ -100,22 +104,34 @@ open class AnvilPlugin : KotlinCompilerPluginSupportPlugin {
                 "'${target.path}'. Only Android and Java modules are supported for now."
         )
       }
+
+      if (!extension.generateDaggerFactories && extension.generateDaggerFactoriesOnly) {
+        throw GradleException(
+            "You cannot set generateDaggerFactories to false and generateDaggerFactoriesOnly " +
+                "to true at the same time."
+        )
+      }
     }
   }
 
   private fun realApply(
     project: Project,
-    isAndroidProject: Boolean
+    isAndroidProject: Boolean,
+    extension: AnvilExtension
```

Comment:
Would it make sense to change the method signature to use the generateDaggerFactoriesOnly boolean instead? You don't seem to use anything else in the AnvilExtension class.

Reply:
I think it's okay the pass the container (the extension) object to the methods.